### PR TITLE
Add duckdb

### PR DESCRIPTION
**Pull request by dcherian**

### CODE REVIEW

**Comment by rabernat** [comment]
Can you do xvec too while we are here?

**Comment by dcherian** [comment]
it's already there
https://github.com/pangeo-data/pangeo-docker-images/blob/4552cca5fcd4377ca3a6806f7cc7ef33d96abd63/pangeo-notebook/environment.yml#L130

File: pangeo-notebook/environment.yml [hunk 1]
```diff
@@ -20,6 +20,7 @@ dependencies:
  - dask-ml
  - datashader
  - descartes
+ - duckdb
```
Comment (weiji14) [inline pangeo-notebook/environment.yml:23]: From reading https://github.com/conda-forge/duckdb-feedstock/issues/6 and https://github.com/conda-forge/python-duckdb-feedstock/pull/105, it seems like `duckdb` is simply an alias for `python-duckdb`, so maybe use `python-duckdb` to be clear that this is only the Python bindings.

Did you want to pull in `duckdb-cli` as well from https://github.com/conda-forge/duckdb-split-feedstock?

Comment (dcherian) [reply pangeo-notebook/environment.yml]: Thanks I added both.

**Review by weiji14 — APPROVED** [review]
Looks ok to me. @scottyhq, do you want to check the docker image size difference?

File: pangeo-notebook/conda-lock.yml [hunk 2]
```diff
@@ -41859,63 +41988,63 @@ package:
...
   hash:
-    md5: a7d4ff4bf1502eaba3fbbaeba66969ec
-    sha256: c8a0c70bb3402b29a9eebb1e41c5d28e9215bb14abea0c986d4d89026aa1ce42
+    md5: 8e25aab3323476d4fd0b5f6bad05d403
+    sha256: 4b0a8143f5c501246214823fe543e9d0749c950fdbdd39de6d8cd6209da2259f
   category: main
   optional: false
 - name: xarray
-  version: 2024.7.0
+  version: 2024.2.0
```
Comment (weiji14) [inline pangeo-notebook/conda-lock.yml:42006]: Oh wait, reminder that we haven't fixed the xarray downgrade issue mentioned at https://github.com/pangeo-data/pangeo-docker-images/pull/577#discussion_r1744592203

**Comment by scottyhq** [comment]
> Looks ok to me. @scottyhq, do you want to check the docker image size difference?

CI log show an increase of ~170MB uncompressed:
```
localhost:5000/pangeo/pangeo-notebook   PR                09fc51b56bcf   21 minutes ago   4.96GB  (from 4.79GB)
```

[Compressed images sizes on dockerhub](https://hub.docker.com/r/pangeo/pangeo-notebook/tags) are currently:
```
85bbcf926278   linux/amd64   1.64 GB
1f6cc912bd9e   linux/arm64   1.56 GB
```

Interestingly on my ARM mac I see a much larger disk usage:
`pangeo/pangeo-notebook   master    419058b9dc6f   3 hours ago     6.72GB`

...So maybe there are some ways to make the ARM images smaller. But we should definitely dig into that separately. Seems like a good addition to me!

**Pull request merged**